libs/vkd3d/command.c

```diff
@@ -4764,27 +4764,32 @@ static void d3d12_command_list_update_dynamic_state(struct d3d12_command_list *l
     /* Make sure we only update states that are dynamic in the pipeline */
     dyn_state->dirty_flags &= list->dynamic_state.active_flags;
 
-    if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_VIEWPORT_COUNT)
+    if (dyn_state->viewport_count)
     {
-        VK_CALL(vkCmdSetViewportWithCountEXT(list->vk_command_buffer,
-                dyn_state->viewport_count, dyn_state->viewports));
-    }
-    else if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_VIEWPORT)
-    {
-        VK_CALL(vkCmdSetViewport(list->vk_command_buffer,
-                0, dyn_state->viewport_count, dyn_state->viewports));
-    }
+        if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_VIEWPORT_COUNT)
+        {
+            VK_CALL(vkCmdSetViewportWithCountEXT(list->vk_command_buffer,
+                    dyn_state->viewport_count, dyn_state->viewports));
+        }
+        else if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_VIEWPORT)
+        {
+            VK_CALL(vkCmdSetViewport(list->vk_command_buffer,
+                    0, dyn_state->viewport_count, dyn_state->viewports));
+        }
 
-    if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_SCISSOR_COUNT)
-    {
-        VK_CALL(vkCmdSetScissorWithCountEXT(list->vk_command_buffer,
-                dyn_state->viewport_count, dyn_state->scissors));
-    }
-    else if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_SCISSOR)
-    {
-        VK_CALL(vkCmdSetScissor(list->vk_command_buffer,
-                0, dyn_state->viewport_count, dyn_state->scissors));
+        if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_SCISSOR_COUNT)
+        {
+            VK_CALL(vkCmdSetScissorWithCountEXT(list->vk_command_buffer,
+                    dyn_state->viewport_count, dyn_state->scissors));
+        }
+        else if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_SCISSOR)
+        {
+            VK_CALL(vkCmdSetScissor(list->vk_command_buffer,
+                    0, dyn_state->viewport_count, dyn_state->scissors));
+        }
     }
+    else
+        FIXME_ONCE("Viewport count is zero. Is this allowed?\n");
```

Comment:
D3D11 allows this; it behaves as if rasterization is disabled so that geometry-only pipelines can still run without viewports. DXVK's not-so-sophisticated way of dealing with this is to bind a dummy viewport with an empty scissor rect; we should probably do the same.

Reply:
I'll do that then, and split it out to a separate PR.

Reply:
Fixed.